### ISSUE TITLE
Plugin state ignores closed nodes

### ISSUE THREAD

**Issue by nytwyk** [vakata/jstree#1012]
Hello,

I am using the plugin "state" to reproduce the current setting of the tree. But the state of a closed node is ignored on reloading the page. All nodes wether they are selected or not are opened. Is this behaviour a fault or wanted?

Best regards
Rainer

**Comment by vakata** [comment]
Here is a demo of the state plugin in action: http://www.jstree.com/demo/
As you can see the file and database demos will always correctly restore their state (selected and opened nodes).
Provided you use the same IDs the state plugin will work fine. If you still have trouble - please provide a demo of the issue so that I can help you.

**Comment by nytwyk** [comment]
Ok, become a little more specific:
If you change your list to plain html (no php or data)
```
<ul>
    <li>root
    <ul>
        <li>Annsi</li>
        <li>fasdf</li>
    </ul></li>
</ul>
```
and reduce your configuration to
```
$('#tree')
    .jstree({
        'plugins' : ['state','checkbox']
    });
```
then the behaviour of jstree become strange (checked closed nodes are open after reload)

Best regards
Rainer

**Comment by vakata** [comment]
Thanks for the clarification - when reporting tickets, try to provide as much details as possible (or a demo).
As for the problem - it is a config option, set it to false in your config (when creating the tree):
http://www.jstree.com/api/#/?q=expand&f=$.jstree.defaults.core.expand_selected_onload

```js
$('#tree')
    .jstree({
        'core' : { expand_selected_onload : false },
        'plugins' : ['state','checkbox']
    });
```

**Comment by nytwyk** [comment]
Hello Ivan Bozhanov,

 

I’ve tried this config option already with no effect. So I’ll send you my complete testing file (a modification of your demo file).

 

Tested with Firefox and Opera

 

Best regards

Rainer Roskothen

 

Von: Ivan Bozhanov [mailto:notifications@github.com] 
Gesendet: Donnerstag, 26. Februar 2015 16:11
An: vakata/jstree
Cc: nytwyk
Betreff: Re: [jstree] Plugin state ignores closed nodes (#1012)

 

Thanks for the clarification - when reporting tickets, try to provide as much details as possible (or a demo).
As for the problem - it is a config option, set it to false in your config (when creating the tree):
http://www.jstree.com/api/#/?q=expand <http://www.jstree.com/api/#/?q=expand&f=%24.jstree.defaults.core.expand_selected_onload> &f=$.jstree.defaults.core.expand_selected_onload

$('#tree')
    .jstree({
        'core' : { expand_selected_onload : false },
        'plugins' : ['state','checkbox']
    });

**Comment by vakata** [comment]
OK, so it has to be something related to the checkbox plugin. I was able to reproduce - thanks.
